spec/controllers/results_controller_spec.rb

```diff
@@ -5,7 +5,8 @@
   let(:student) { create :student, grace_credits: 2 }
   let(:instructor) { create :instructor }
   let(:ta) { create :ta }
-  let(:grouping) { create :grouping_with_inviter, assignment: assignment, inviter: student }
+  let(:grouping) { create :grouping_with_inviter, assignment: assignment, inviter: student, tas: [ta] }
```

Comment:
You shouldn't add the `tas` association here. Instead, in the `context`s below where you want the ta to be associated with the grouping, use

```
let!(:ta_membership) { create :ta_membership, role: ta, grouping: grouping }
```

(You don't need the `let` on line 9, since it isn't being used directly. The `let!` will eagerly evaluate the membership so that the TA is correctly assigned to the grouping.)